### ISSUE TITLE
SolrNet.Autofac.0.5.0-alpha2 Nuget install failed

### ISSUE THREAD

**Issue by gopala00** [SolrNet/SolrNet#217]
I'm using AutoFac 3.5.2. When I installed SolrNet.Autofac.0.5.0-alpha2 Nuget package I get the error:
Install-Package : Unable to find a version of 'Autofac' that is compatible with 'SolrNet.Autofac 0.5.0-alpha2 constraint: Autofac (= 2.6.3.862)

I tried adding the following to app.config, but still cannot work around.

```
      <dependentAssembly>
        <assemblyIdentity name="Autofac" publicKeyToken="17863af14b0044da"/>
        <bindingRedirect oldVersion="0.0.0.0-3.5.2.0" newVersion="3.5.2.0"/>
      </dependentAssembly>
```

Appreciate if anyone suggest a work around.

**Comment by mausch** [comment]
I don't know what kind of breaking changes there are between 2.6.3 and 3.5.2. If you're interested in upgrading this dependency could you please send a pull request?

**Comment by Mykezero** [comment]
Autofac is using DNX targeting the .Net framework 4.5. Would there be any problems if AutofacContrib projects are upgraded to use this version of the framework? I know the Autofac nuget package will install under that version of the framework. 

Also, what do you use to run the tests for this project? I can't seem to find how to run them.

**Comment by lvanderbijl** [comment]
Hi All,

I was able to download and upgrade the project to the latest autofac version though I too had to upgrade to .net 4.5.

I also struggled to get mbunit to work so converted the tests to mstest and found all the tests passed succesfully.

I didn't want to submit a pull request since I had to upgrade the .net version and create a new mstest project.

But it does seem to work without trouble.

**Comment by alex-training** [comment]
Hi @mausch 
Is there any progress on this?

I want to use your package (for Solr4) but I can't do this because of strict dependencies on very old Autofac version.
The same problem is with CommonServiceLocator.

Can you do something please?
I would suggest to use more soft dependencies (f.e >= [some version])

**Comment by gjunge** [comment]
Fixed by #286

**Comment by xavier-morera** [comment]
Released version 0.7.1
